src/mctp-bench.c

```diff
@@ -116,6 +121,12 @@ static int handle_incoming_msg(struct recv_ctx *recv_ctx)
         }
 
         hdr = (struct msg_header *)recv_ctx->buf;
+        if (memcmp(hdr->vendor_prefix, VENDOR_TYPE_BENCH, sizeof(VENDOR_TYPE_BENCH)) != 0) {
```

Comment:
We should check that we have at least `sizeof(VENDOR_TYPE_BENCH)`.

But I guess we're not checking that we have a full header either; we should probably add that, before the initial `(struct msg_header *)` cast.

Reply:
I've added a check.